### PR TITLE
Test disable polling enabled does not sync Gitrepo (Fleet-124)

### PR DESCRIPTION
**Pull request by mmartin24**
Adds test https://app.qase.io/case/FLEET-124

## Overview:
Implementend test to ensure that when flag `disablePolling` is set to `True`, changes done on Github repo do not affect Gitrepo even after seconds

## Done
- Added test
- Added bash scripts to change replicas over Github repo on a given moment through `cy.exec`
- Added `cy.task ` to output log in console
- Encapsulated yaml addition into new function `addYamlFile`

## Tests
Local test passing. Video

https://github.com/user-attachments/assets/0839adfa-8db3-4a94-af0f-065ffc738ce6

o:

### CODE REVIEW

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts [hunk 1]
```diff
@@ -742,3 +742,88 @@ describe("Test Application deployment based on 'clusterSelector'", { tags: '@p1'
...
+        git add nginx.yaml
+  
+        if ! git diff --quiet origin/main; 
+        then
+          echo "Changes detected from original repo. Changing replicas to 5"
+          git commit -m 'Ensuring initial number of replicas is 5'
+          git push -u origin main
+        else
+          echo -e "Nothing to push, done"
+        fi
```
Comment (sbulage) [inline tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:816]: Can this be put in bash file and execute it?

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts [hunk 2]
```diff
@@ -742,3 +742,88 @@ describe("Test Application deployment based on 'clusterSelector'", { tags: '@p1'
...
+      test -e "$PWD/fleet-qa-examples-public/disable-polling"
+      if [ $? -ne 0 ]; then
+        echo "Error: path $PWD/fleet-qa-examples-public/disable-polling does not exist"
+        exit 1
+      fi
+
+      echo -e "Ensuring replicas is 2"
+      sed -i 's/replicas: ..*/replicas: 2/g' $PWD/fleet-qa-examples-public/disable-polling/nginx.yaml
+
+      echo -e "Commiting and pushing if needed "
```
Comment (sbulage) [inline tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:767]: Why `pushing if needed`  please elaborate more in the same `echo`?

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts (hunk 1 above)
Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:816]: mmm. Good point. I will see if this is possible something similar to this: https://stackoverflow.com/questions/76965361/can-cypress-execute-a-shell-script-in-an-existing-terminal-window
Thanks

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts (hunk 2 above)
Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:767]: By 'if needed' I meant if number of replicas is different from initially pre-established.
Ok, let me rephrase that.
Thanks.

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts (hunk 1 above)
Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts]: Done. thanks a lot for the suggestion

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts (hunk 2 above)
Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts]: Done, thanks a lot for the suggestion

File: tests/assets/disable_polling_reset_2_replicas.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+echo -e "Cloning repo"
```
Comment (sbulage) [inline tests/assets/disable_polling_reset_2_replicas.sh:3]: Suggestion: Add `\n` in every `echo`. The output in the `cy.log` is not formatted. <-- May be you can implement this suggestion in next PR with similar test.

File: tests/assets/disable_polling.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+apiVersion: fleet.cattle.io/v1alpha1
+kind: GitRepo
+metadata:
+  name: test-disable-polling
+  annotations:
+    {}
+    #  key: string
+  labels:
+    {}
+    #  key: string
```
Comment (sbulage) [inline tests/assets/disable_polling.yaml:10]: This is not required. Also, on line no. 14-15 and line no. 9 as well. <-- Suggestion can be implemented in next PR.

**Comment by sbulage** [comment]
File name: `tests/assets/disable_polling_reset_2_replicas.sh` --> `tests/assets/disable_polling_with_default_2_replicas.sh`
reset seems like you're changing replicas back to 2.

Not blocker for this PR :smile:

**Review by sbulage — APPROVED** [review]
LGTM :+1: wating for CI :green_circle:

File: tests/assets/disable_polling_reset_2_replicas.sh (hunk 3 above)
Comment (mmartin24) [reply tests/assets/disable_polling_reset_2_replicas.sh:3]: Thanks, but that was the intention with `echo -e`. It does not format it anyway there.

File: tests/assets/disable_polling.yaml (hunk 4 above)
Comment (mmartin24) [reply tests/assets/disable_polling.yaml]: Changes applied.
Removed also commented `-string`line.
I left however 
` correctDrift:
    enabled: false` given that by default it was created in Yaml when playing it and I am afraid it may take the true value. It can be tweaked later

**Comment by mmartin24** [comment]
> File name: `tests/assets/disable_polling_reset_2_replicas.sh` --> `tests/assets/disable_polling_with_default_2_replicas.sh` reset seems like you're changing replicas back to 2.
> 
> Not blocker for this PR 😄

hey, that is exactly what I mean. Resetting to original value 2
